pkg/controller/bookkeepercluster/bookie.go

```diff
@@ -365,29 +401,106 @@ func createVolumeMount(ledgerDirs []string, journalDirs []string, indexDirs []st
 }
 
 func makeBookieVolumeClaimTemplates(bk *v1alpha1.BookkeeperCluster) []corev1.PersistentVolumeClaim {
-	return []corev1.PersistentVolumeClaim{
-		{
+	var ledgerDirs, journalDirs, indexDirs []string
+	var multiVolumesSetPerBookieEnabled bool
+	var ok bool
+
+	if _, ok = bk.Spec.Options["ledgerDirectories"]; ok {
```

Comment:
Lines 407-428 might not be required, if we are making multivolume support  to `false` . Could you please verify that?

Reply:
> Lines 407-428 might not be required, if we are making multivolume support to `false` . Could you please verify that?

OK. I will optimize this part according to your suggestion.